src/main/java/seedu/duke/commands/DeleteAccountCommand.java

```diff
@@ -37,21 +42,41 @@ private void processCommand() throws InvalidDeleteAccountCommandException {
 
     }
 
-    private void printSuccess(Ui ui) {
+    private void printSuccess (Ui ui) {
         ui.printf(Message.SUCCESSFUL_DELETE_ACCOUNT_COMMAND.getMessage(), this.currency.name());
     }
 
+    /**
+     * delete all related currency transaction in the main transaction list
+     *
+     * @param currency currency to be delete
+     */
+
+    private void deleteCurrencyTransaction (Currency currency) {
+        ArrayList<Transaction> transactionList = transactions.getTransactionList();
```

Comment:
We should rethink this portion. I think it may be better to move this function into the TransactionManager itself so it can access the list directly.

Reply:
You are right, I will change this